iris/sentryiris_test.go

```diff
@@ -0,0 +1,479 @@
+		// {
+		// 	RequestPath: "/404/1",
+		// 	RoutePath:   "",
+		// 	Method:      "GET",
+		// 	WantStatus:  404,
+		// 	Handler:     nil,
+		// 	WantTransaction: &sentry.Event{
+		// 		Level:       sentry.LevelInfo,
+		// 		Type:        "transaction",
+		// 		Transaction: "GET /404/1",
+		// 		Request: &sentry.Request{
+		// 			URL:    "/404/1",
+		// 			Method: "GET",
+		// 			Headers: map[string]string{
+		// 				"Accept-Encoding": "gzip",
+		// 				"User-Agent":      "Go-http-client/1.1",
+		// 			},
+		// 		},
+		// 		TransactionInfo: &sentry.TransactionInfo{Source: "url"},
+		// 		Extra:           map[string]interface{}{"http.request.method": string("GET")},
+		// 	},
+		// 	WantEvent: nil,
+		// },
+		{
+			RequestPath: "/post",
+			RoutePath:   "/post",
+			Method:      "POST",
+			WantStatus:  200,
+			Body:        "payload",
+			Handler: func(ctx iris.Context) {
+				hub := sentryiris.GetHubFromContext(ctx)
+				body, err := io.ReadAll(ctx.Request().Body)
+				if err != nil {
+					t.Error(err)
+				}
+				hub.CaptureMessage("post: " + string(body))
+				ctx.StatusCode(http.StatusOK)
+				_ = ctx.JSON(map[string]any{"status": "ok"})
+			},
+			WantTransaction: &sentry.Event{
+				Level:       sentry.LevelInfo,
+				Type:        "transaction",
+				Transaction: "POST /post",
+				Request: &sentry.Request{
+					URL:    "/post",
+					Method: "POST",
+					Data:   "payload",
+					Headers: map[string]string{
+						"Content-Length":  "7",
+						"Accept-Encoding": "gzip",
+						"User-Agent":      "Go-http-client/1.1",
+					},
+				},
+				TransactionInfo: &sentry.TransactionInfo{Source: "route"},
+				Extra: map[string]interface{}{
+					"http.request.method":       string("POST"),
+					"http.response.status_code": int(200),
+				},
+			},
+			WantEvent: &sentry.Event{
+				Level:   sentry.LevelInfo,
+				Message: "post: payload",
+				Request: &sentry.Request{
+					URL:    "/post",
+					Method: "POST",
+					Data:   "payload",
+					Headers: map[string]string{
+						"Accept-Encoding": "gzip",
+						"Content-Length":  "7",
+						"User-Agent":      "Go-http-client/1.1",
+					},
+				},
+			},
+		},
+		{
+			RequestPath: "/get",
+			RoutePath:   "/get",
+			Method:      "GET",
+			WantStatus:  200,
+			Handler: func(ctx iris.Context) {
+				hub := sentryiris.GetHubFromContext(ctx)
+				hub.CaptureMessage("get")
+				ctx.StatusCode(http.StatusOK)
+				_ = ctx.JSON(map[string]any{"status": "get"})
+			},
+			WantTransaction: &sentry.Event{
+				Level:       sentry.LevelInfo,
+				Type:        "transaction",
+				Transaction: "GET /get",
+				Request: &sentry.Request{
+					URL:    "/get",
+					Method: "GET",
+					Headers: map[string]string{
+						"Accept-Encoding": "gzip",
+						"User-Agent":      "Go-http-client/1.1",
+					},
+				},
+				TransactionInfo: &sentry.TransactionInfo{Source: "route"},
+				Extra: map[string]interface{}{
+					"http.request.method":       string("GET"),
+					"http.response.status_code": int(200),
```

Comment:
Nitpick, but what's with all these conversions to int? And for all other integrations I used constants from net/http, perhaps we could use them here as well (i.e. http.StatusOK for this one).

Reply:
In fact you used them in your PR for Negroni, so should put them here too.